### PR TITLE
correct varchar() binary 

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve? 
Refer to https://github.com/pingcap/tidb/pull/20269

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

File: ast/ddl_test.go [hunk 1]
```diff
@@ -260,7 +260,7 @@ func (ts *testDDLSuite) TestDDLColumnDefRestore(c *C) {
 		{"id DATETIME", "`id` DATETIME"},
 		{"id DECIMAL(4,2)", "`id` DECIMAL(4,2)"},
 		{"id char(1)", "`id` CHAR(1)"},
-		{"id varchar(10) BINARY", "`id` VARCHAR(10) BINARY"},
```
Comment (kennytm) [inline ast/ddl_test.go:263]: this looks wrong to me.

Comment (wjhuang2016) [reply ast/ddl_test.go:263]: Yeah, this PR is not complete.
I'm thinking that how to deal with BINARY.
Ignore it or set collation like MySQL?
I prefer to ignore it.

Comment (tangenta) [reply ast/ddl_test.go:263]: @nullnotnil PTAL

Comment (nullnotnil) [reply ast/ddl_test.go:263]: I prefer set collate over ignoring. Ignoring could create problems if collations are enabled and there is an index on the column.

As @kennytm points out, `BINARY` is just an alias for collate as the current charsets binary collation. It even does this in MySQL 5.7, but without the deprecation warning:

```sql
mysql [localhost:5731] {msandbox} (test) > CREATE TABLE t (a varchar(10) binary);
Query OK, 0 rows affected (0.01 sec)

mysql [localhost:5731] {msandbox} (test) > SHOW CREATE TABLE t\G
*************************** 1. row ***************************
       Table: t
Create Table: CREATE TABLE `t` (
  `a` varchar(10) CHARACTER SET latin1 COLLATE latin1_bin DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1
1 row in set (0.00 sec)
```

**Comment by wjhuang2016** [comment]
After the discussion, this problem should be fixed in pingcap/tidb#20269.

**Pull request closed without merging**